File: abc/abc233/c.py
def count(idx: int, prod: int, a: list, x: int) -> int:
    if idx == len(a):
        return int(prod == x)
    ans = 0
    for num in a[idx]:
        ans += count(idx + 1, prod * num, a, x)
    return ans


def main():
    n, x = map(int, input().split())
    a = [list(map(int, input().split()))[1:] for _ in range(n)]
    print(count(0, 1, a, x))


if __name__ == '__main__':
    main()

File: abc/abc233/c.cpp
#include <iostream>
#include <limits>
#include <vector>

int count(int idx, long long prod, const std::vector<std::vector<long long>> a, long long x) {
    if (idx == static_cast<int>(a.size())) {
        return static_cast<int>(prod == x);
    }
    int ans = 0;
    for (const auto &num : a[idx]) {
        if ((std::numeric_limits<long long>::max() / prod) < num) {
            continue;
        }
        ans += count(idx + 1, prod * num, a, x);
    }
    return ans;
}

int main() {
    int n;
    long long x;
    std::cin >> n >> x;
    std::vector a(n, std::vector<long long>());
    for (auto &ai : a) {
        int l;
        std::cin >> l;
        ai.resize(l);
        for (auto &aj : ai) {
            std::cin >> aj;
        }
    }

    std::cout << count(0, 1, a, x) << std::endl;
    return 0;
}